onnxruntime/core/providers/dml/DmlExecutionProvider/src/AbiCustomRegistry.h

```diff
@@ -38,11 +38,12 @@ class AbiCustomRegistry : public WRL::Base<IMLOperatorRegistry, IMLOperatorRegis
         _In_opt_ IMLOperatorShapeInferrer* shapeInferrer,
         _In_opt_ IMLOperatorSupportQueryPrivate* supportQuery,
         bool isInternalOperator,
-        bool canAliasFirstInput,
         bool supportsGraph,
         const uint32_t* requiredInputCountForGraph = nullptr,
         _In_reads_(constantCpuInputCount) const uint32_t* requiredConstantCpuInputs = nullptr,
-        uint32_t constantCpuInputCount = 0) const noexcept override;
+        uint32_t constantCpuInputCount = 0,
+        _In_reads_(aliasCount) const std::pair<uint32_t, uint32_t>* aliases = nullptr,
```

Comment:
Hmm, if this was truly a COM interface, then we wouldn't accept any `std` types across the ABI, but I believe this is actually purely an internal interface anyway. So, then if we *can* use `std` types, then why not use `std::span` (or `gsl::span`) rather than a pointer+count?

`const std::span<const std::pair<uint32_t, uint32_t>> aliases`